Controllers films users

KoRVeRT opened the pull request:
Добавлено два класса - Film и User.
Добавлено два класса-контроллера - FilmController и UserController.

KoRVeRT committed:
Initial version of the Sprint 9 task

KoRVeRT committed:
Working version №1

KoRVeRT committed:
Final version Sprint 9

kalitva commented on pom.xml:
```diff
@@ -0,0 +1,59 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
```
В проекте должны использоваться либо везде табы, либо везде пробелы, иначе будет путаница с отступами. Замени все табы на пробелы

В intellij нажми Edit --> Convert Indents --> To spaces

kalitva commented on pom.xml:
```diff
@@ -0,0 +1,59 @@
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.7.8</version>
+		<relativePath/> <!-- lookup parent from repository -->
```
Удали коммент

kalitva commented on src/main/java/ru/yandex/practicum/filmorate/FilmorateApplication.java:
```diff
@@ -0,0 +1,13 @@
+@SpringBootApplication
+public class FilmorateApplication {
+
+	public static void main(String[] args) {
```
Здесь тоже замени табы на пробелы

kalitva commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,49 @@
+    private int id;
+    private final HashMap<Integer, Film> films = new HashMap<>();
+
+    @GetMapping()
```
Если аргументы не передаются, скобки для аннотаций не ставятся

kalitva commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,28 @@
+public class Film {
+    int id;
+    @NotBlank
+    String name;
```
Лучше отделять поля с аннотациями пустой линией, так легче читается когда много полей

kalitva commented on src/main/java/ru/yandex/practicum/filmorate/valid/MovieReleaseDate.java:
```diff
@@ -0,0 +1,24 @@
+@Documented
+public @interface MovieReleaseDate {
+
+    String message() default "дата выхода — не ранее 28 декабря 1895 года";
```
Сообщение для исключения - по английски

KoRVeRT committed:
Заменить tab на пробелы.
Удалить комментарий 9 строка в pom.xml.
Добавить пробел между полями с аннотацией в классах Film и User.
MovieReleaseDate сообщение исключения изменить на английский язык.
Убрать скобки у аннотаций без аргументов FilmController и UserController.

kalitva commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,42 @@
+@EqualsAndHashCode
+@ToString
+public class User {
+    int id;
```
Поля-то приватными же должны быть

kalitva commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,42 @@
+        this.id = id;
+        this.login = login;
+        this.email = email;
+        if (name == null || name.isBlank()) {
```
В следующем спринте перенеси эту логику в сервис

KoRVeRT committed:
Добавить private к полям классов User и Film
